### PR TITLE
Implement new schema types and support versioned schemas

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose/Changes

- Support the new schema type: ULeb128, ILeb128, ByteArray and ByteList.
- Support schema modules with versioning.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/common/src/serializationHelpers.ts [hunk 1]
```diff
@@ -791,6 +849,96 @@ export function serializeSchemaFields(
...
+        buffer.push(encodeWord8(lastByte ? byte : byte | 0b1000_0000));
+        if (lastByte) {
+            return Buffer.concat(buffer);
+        }
+    }
+    throw new Error('Invalid LEB128 unsigned integer encoding');
+}
+
+/**
+ * Serialize a string with signed integer using LEB128 to a Buffer
```
Comment (Bargsteen) [inline packages/common/src/serializationHelpers.ts:888]:
```suggestion
 * Serialize a string with a signed integer using LEB128 to a Buffer.
```

File: packages/common/src/types.ts [hunk 2]
```diff
@@ -1338,6 +1338,14 @@ export enum ParameterType {
     ContractName,
     /** Receive function name. */
     ReceiveName,
+    /** LEB128 encoding of an unsigned integer */
+    ULeb128,
+    /** LEB128 encoding of an signed integer */
```
Comment (Bargsteen) [inline packages/common/src/types.ts:1343]:
```suggestion
    /** LEB128 encoding of a signed integer */
```

File: packages/common/test/deserializeSchema.test.ts [hunk 3]
```diff
@@ -31,3 +31,32 @@ test('Deserialize V2 schema from file (icecream)', async () => {
...
+    expect(parsedModule.value[contractName].init).toBeDefined();
+    expect(parsedModule.value[contractName].receive).toBeDefined();
+});
+
+test('Deserialize V2 versioned schema from file (cis2-wccd)', async () => {
+    const rawModule = Buffer.from(
+        fs.readFileSync('./test/resources/cis2-wccd-schema-v2-versioned.bin')
+    );
+    const parsedModule = deserialModuleFromBuffer(rawModule);
+
```
Comment (Bargsteen) [inline packages/common/test/deserializeSchema.test.ts:57]: Please add a check to ensure that this is a v2 schema. (Similar to the v1 test above)

**Review by Bargsteen — COMMENTED** [review]
Nice work. I just have a few minor issues and concerns 😊

File: packages/common/src/deserializeSchema.ts [hunk 4]
```diff
@@ -61,6 +61,40 @@ export function deserialModuleV2(source: Readable): ModuleV2 {
     )(source);
 }
 
+/** Magic prefix for versioned schemas, used to distinguish between a versioned schema and the older format without versioning as part of the schema */
+export const VERSIONED_SCHEMA_PREFIX = 65535; // The constant is corresponding to two full bytes.
```
[the remark this answers is not in the document]
Comment (limemloh) [reply packages/common/src/deserializeSchema.ts]: I have changed it to use `maxed-out bytes` instead

**Pull request merged**